src/java/org/apache/cassandra/db/guardrails/Values.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.db.guardrails;
+
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+
+import com.google.common.collect.Sets;
+
+import org.apache.cassandra.service.ClientState;
+
+import static java.lang.String.format;
+
+/**
+ * A guardrail that warns about but ignores some specific values, and rejects the use of some other values.
+ *
+ * @param <T> The type of the values of which certain are disallowed.
+ */
+public class Values<T> extends Guardrail
+{
+    /*
+     * Implementation note: as mentioned in the class Javadoc and for consistency with the other Guardrail
+     * implementation of this class (and to generally avoid surprises), this implementation ensures that live
+     * changes to the underlying guardrail setting gets reflected. This is the reason for the relative
+     * "complexity" of this class.
+     */
+
+    private final Function<ClientState, Set<String>> ignoredRaw;
+    private final Function<ClientState, Set<String>> disallowedRaw;
+    private final Function<String, T> parser;
+    private final String what;
+
+    private volatile Set<T> cachedIgnoredValues;
+    private volatile Set<T> cachedDisallowedValues;
+    private volatile Set<String> cachedIgnoredRaw;
+    private volatile Set<String> cachedDisallowedRaw;
```

Comment:
Can we remove?  We can keep this simple and remove `parser`.

The only usage is

```
Guardrails.allowedTableProperties.guard(attrs.updatedProperties(), attrs::removeProperty, state);
```

and all properties are lower case, so do we want to really support someone configuring with mixed or upper case if it makes us deal with costly state?

Reply:
Also, this doesn't work if you are going to have different configs for different users, as you share the same cache; so this will just cause race conditions and wrong configs going to different users

Reply:
also looks like most of the complexity of this class is coming from this feature to lowercase + cache...

Reply:
That's meant to be used also for parsing properties that are not a set of `String`, like having a guardrail of disallowed consistency levels. In that case the type of the guardrail would be `Values<ConsistencyLevel>`. Indeed the current caching mechanism wouldn't work with the current approach to pluggability. What we could do instead is moving the parsing to the `GuardrailsConfig` objects, so the config reads the values from the yaml as strings, and have a getter returning `Set<ConsistencyLevel>`. That `Set<ConsistencyLevel>` could be stored on the `GuardrailsConfig` instance so we don't have to parse it on every call to the getter.

Nevertheless, we can go with the simpler `Values` guardrail implementation without parsing nor caching in this prototype, and re-evaluate when we add a guardrail for something like allowed consistency levels.

Reply:
> and re-evaluate when we add a guardrail for something like allowed consistency levels.

Lets go simple for now.  Even in your example as we are mostly checking for set contains, enums work fine with sets so removing the idea of parsing in favor of genetics also solves this problem without adding race condition concerns.